lasy/utils/laser_utils.py

```diff
@@ -211,6 +206,78 @@ def get_full_field(laser, theta=0, slice=0, slice_axis="x", Nt=None):
     return env, ext
 
 
+def get_spectrum(grid, dim, bins=20, range=None, omega0=None, phase_unwrap_1d=None):
```

Comment:
Just a comment here: the spectrum could also be obtained from a Fourier transform, which would work very easily with field or envelope (in the latter case, one would just need to add `omega0` to the spectrum) and could be more accurate, so it will be very interesting to compare both. I am sure @hightower8083 would share this view :)

But this is definitely not required in this PR, can be added later.

Reply:
I agree. It would be nice to support several methods here as well as in other utils (e.g., `get_duration`` could have RMS, FWHM, fit, etc. options).